### ISSUE TITLE
Bluetooth headset not ringing 

### ISSUE THREAD

**Issue by torxt** [signalapp/Signal-Android#9727]
- [x ] I have searched open and closed issues for duplicates
- [ x] I am submitting a bug report for existing functionality that does not work as intended
- [ x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x ] This isn't a feature request or a discussion topic

There is a duplicate of this bug in the ios app but I did not find one for Android. 

Android 9. Lenovo phone
I don't get get Headset to ring. The phone vibrates but no ringing in the Bluetooth headset. Calling via Sim works with and without Bluetooth. 

When I accept the call via signal sound is handled by the headset. Only the call sound is mute. It's is a kind of new but it did work about a month ago. I did not update the phone firmware.

----------------------------------------

### Bug description
When phone is connected to Bluetooth headset, the phone does vibrate but no calling sound is played via Bluetooth headset. If headset is disconnected, ring sound plays normal 

### Steps to reproduce 
- connect Bluetooth headset to phone 
- get called 

**Actual result:** 
- phone will vibrate but not play ring sound via headset 
**Expected result:** 
- play ring sound via headset so you will know that you have been called 

### Device info

**Device:** lenovo p2
**Android version:** 9
**Signal version:** 4.62. 4

### Link to debug log

https://debuglogs.org/4ce3e33dba9408009c0618c2bbad0e19b838db959cb2f061548ac6530b501f40

**Comment by dpociu** [comment]
I can confirm the same behavior on Samsung Galaxy S10 (SM-G973U1) , Android 10 Baseband Version G973U1UES4DTH3

**Comment by charltonstanley** [comment]
I actually don't have this problem on Android 10. Notification sounds for signal messages and phone calls work just fine via Bluetooth headset.

Phone: Pixel 3
Baseband Version: g845-00107-200409-B-6382632
Build Number: QQ3A.200805.001
Bluetooth Headset: Sony WH-H900N

**Comment by BurhanBudak** [comment]
I can also confirm this bug.

Phone: Oneplus 3T
Bluetooth Headset: Oneplus Bullets Wireless 2

**Comment by Trojgarb** [comment]
I can confirm this. No ringing with bluetooth headset connected. Moto X Force, stock Android 7.0, Signal 5.4.12

**Comment by ET0269** [comment]
I can confirm the same behavior on Samsung Galaxy S10 (SM-G973W) , Android 11 Baseband Version G973WVLS4FUB1 Android Security Patch  Feb. 1/21

**Comment by Trojgarb** [comment]
After update to version 5.5.5 still no sound of ringing with bluetooth headset connected
Phone: Moto X Force
System: stock Android 7.0
Signal Version: 5.5.5
Bluetooth Headset: Sony WH-CH510

**Comment by ET0269** [comment]
There must be another thread somewhere?? As I can't believe this issue is not being worked on???

**Comment by UserX404** [comment]
@Trojgarb 
This issue isn't neither addressed in 5.4.x nor in 5.5.x. Please do not bump threads and use the [community forum](https://community.signalusers.org/) for discussion.
Thank you!

**Comment by ET0269** [comment]
I can confirm not only the same behavior is still occurring, but after installing 5.5.5 on Samsung Galaxy S10 (SM-G973W) , it now doesn't even vibrate, so no notification at all when receiving a phone call, only afterwards I get a missed call notification. Android 11 Baseband Version G973WVLS4FUB1 Android Security Patch Feb. 1/21

**Comment by Siron777** [comment]
Same issue on a Samsung Galaxy A 51 with android 11 and Signal 5.8.10, the phone is vibrating, but I do not hear the ringing in the headset, nor in the phone.

**Comment by ET0269** [comment]
I have battled with this for years. It's a joke, this whole support line is BS. They never ever get back or give a response. Years and years so bad I finally said screw it and went back to Whats App, and I HATE the fact I had to do that. So many people I moved to Signal 'cause ya know, it's better etc etc blah blah blah. I looked like a total fool. Every phone from every other person had the same issue. It's a joke.

**Comment by Siron777** [comment]
Still same issue for me. Notification are problematic with Signal (Bluetooth or event without it). I really like Signal, but I am also a lot less enthusiast to promote it due to that.

**Comment by ET0269** [comment]
Yes, save yourself the pain and DO NOT promote it, as you will have egg on your face. Signal could be great, however when they go years with out fixing such an important issue....or even try to address it and just blatantly dismiss it. Give me a break. Like no one on the planet uses a Bluetooth headset. Such a joke.

**Comment by jknockaert** [comment]
This may be caused by Signal using `MediaPlayer` for incoming calls ringtone, as described in #11512.

**Comment by ET0269** [comment]
@jknockaert Yes, I saw this thanks, but I do not believe I use MediaPlayer for incoming calls ringtone

**Comment by jknockaert** [comment]
@ET0269 The use of the `MediaPlayer` API is hardcoded in Signal. Actually a notification arrives and then Signal starts playing the ringtone using the `MediaPlayer` API. That API is actually meant for playing audio and video, not for ringtones. So the Android settings (such as volume) for media apply to Signal's ringtone.

**Comment by ET0269** [comment]
@jknockaert Yes, that's what I thought. So what is the work around or solution, as Signal calls/notifications never ring through to the bluetooth headset. Just want to give you a big THANKYOU as this is the most attention I have ever received in regards to this issue in the last 2 Yrs.

**Comment by jknockaert** [comment]
@ET0269 There's no work around. I'm not sure why the `MediaPlayer` API was selected for playing ringtone. There may have been a valid reason; not sure if it still applies to more modern versions of Android. The only thing that can be done is prepare a PR to fix the code and hope that it gets merged. But I have no experience with coding for Android so I prefer to leave this to the devs.

**Comment by ET0269** [comment]
@jknockaert Seems to me you know more of this issue then what the devs are giving this issue. I'm sure you have passed on your thoughts and they know, yet they (the devs) chose and choose to pay more attention to emoji's and stickers. Again thanks for all your insite.

**Comment by sellth** [comment]
There's a thread in the community forum about the call API thing, maybe give it a look: https://whispersystems.discoursehosting.net/t/implement-android-telecom-api-for-calls/41280

**Comment by cody-signal** [comment]
Try on 5.33 where we've integrated with the Android telecom system, and made some changes to how we ring. Please post an updated debuglog if you try with that version and still encounter problems.

**Comment by charltonstanley** [comment]
>Try on 5.33 where we've integrated with the Android telecom system, and made some changes to how we ring. Please post an updated debuglog if you try with that version and still encounter problems.
Hey @cody-signal, when will `5.33` be released to the Play Store? The Play Store still shows `5.32.15` as the latest version.

**Comment by charltonstanley** [comment]
>Try on 5.33 where we've integrated with the Android telecom system, and made some changes to how we ring. Please post an updated debuglog if you try with that version and still encounter problems.

Hey @cody-signal, when will `5.33` be released to the Play Store? The Play Store still shows `5.32.15` as the latest version.

**Comment by cody-signal** [comment]
Soon, starting a prod rollout today, so likely within the next few days, or sooner if you get lucky and included in the partial rollouts.

**Comment by point-source** [comment]
Still having this issue on 5.36.3

**Comment by cody-signal** [comment]
Thanks, this does get fixed by the telecom integration, however, the integration causes other issues we need to resolve before we can roll it out widely again.

**Comment by point-source** [comment]
Thanks for the update! That's great news that there is a known fix.
